Lib/test/test_unicode.py

```diff
@@ -185,6 +185,18 @@ def test_iterators(self):
         self.assertEqual(next(it), "\u3333")
         self.assertRaises(StopIteration, next, it)
 
+    def test_iterators_invocation(self):
+        cases = [type(iter('abc')), type(iter('🚀'))]
+        for cls in cases:
+            with self.subTest(cls=cls):
+                self.assertRaises(TypeError, cls)
+
+    def test_iteration(self):
+        cases = ['abc', '🚀🚀🚀', "\u1111\u2222\u3333"]
+        for case in cases:
+            with self.subTest(string=case):
+                self.assertEqual((*case,), (*iter(case),))
```

Comment:
Both sides here use iteration, so this doesn't actually test that it works correctly. Use `"".join(iter(case))` on the RHS instead.

Reply:
I was waiting for buildbots to finish before committing more changes, I push the fix now.